app/controlplane/integrations/dependencytrack/cyclonedx/v1/integration.go

```diff
@@ -26,56 +26,59 @@ import (
 	pb "github.com/chainloop-dev/chainloop/app/controlplane/integrations/gen/dependencytrack/cyclonedx/v1"
 	"github.com/chainloop-dev/chainloop/app/controlplane/integrations/sdk/v1"
 	"github.com/go-kratos/kratos/v2/log"
-	"google.golang.org/protobuf/types/known/anypb"
 )
 
 const ID = "dependencytrack.cyclonedx.v1"
-const version = "1.0"
-const description = "Dependency Track CycloneDX Software Bill Of Materials Integration"
-
-var _ sdk.FanOut = (*DependencyTrack)(nil)
 
 type DependencyTrack struct {
-	*sdk.BaseIntegration
+	*sdk.FanOutIntegration
+}
+
+type registrationConfig struct {
```

Comment:
The idea here is to decouple the `API schema` (currently defined and validated with protobuffers but in the future it might be defined differently) from the `state` that an extension author wants to preserve during the life-cycle of the extension.

It's true that today both the API and the state looks pretty much identical but it does not need to be like that. In fact if you look closely, the `api key` is not in the state. So credentials are handled differently. 

The other reason for maintaining your own structs is to make sure the plugin is decoupled from proto messages, for the most part.

Before this change, both the state and the API was defined in the proto files, what do you think, should we keep it like before, btw. note, nothing prevents an author to define and re-use structs from protos.

Reply:
Makes sense! Thanks for the detailed explanation.